SYCL/ESIMD/api/functional/shared_elements.hpp

```diff
@@ -0,0 +1,43 @@
+//===-- shared_elements.hpp - Function that provides USM with a smart pointer.
+//      -------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides data struct that lets interact with USM with a smart
+/// pointer that lets avoid memory leaks.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <sycl/sycl.hpp>
+
+namespace esimd_test::api::functional {
+
+template <typename T> class shared_elements {
+  std::unique_ptr<T, std::function<void(T *)>> m_allocated_data;
+
+public:
+  shared_elements(sycl::queue &queue, size_t num_elements = 1) {
```

Comment:
OK, I see now, it's just another way to define `shared_vector` of a fixed size. I was talking about scalar variables to get rid of `operator[]` though..

Reply:
I've done it [here](https://github.com/intel/llvm-test-suite/pull/762/commits/afbf175e94a84d6a621ba620d2c0119abd0ec6aa#diff-9d2d5320f066e81e77b17efb5f2888df4d6d5e653f591eb2e9d39f34c267172cL40-R42)